worlds/faxanadu/Locations.py

```diff
@@ -0,0 +1,199 @@
+from typing import List
+
+
+class LocationType():
+    world = 1 # Just standing there in the world
+    hidden = 2 # Kill all monsters in the room to reveal, each "item room" counter tick.
+    boss_reward = 3 # Kill a boss to reveal the item
+    shop = 4 # Buy at a shop
+    give = 5 # Given by an NPC
+    spring = 6 # Activatable spring
+    boss = 7 # Entity to kill to trigger the check
+
+
+class ItemType():
+    unknown = 0 # Or don't care
+    red_potion = 1
+
+
+class LocationDef:
+    def __init__(self, id: int, name: str, region: str, type: LocationType, original_item: ItemType):
```

Comment:
```suggestion
    def __init__(self, id: Optional[int, None], name: str, region: str, type: int, original_item: int):
```
These end up being ints rather than their class types